jina/optimizers/__init__.py

```diff
@@ -140,14 +141,57 @@ def _trial_parameter_sampler(self, trial):
         trial_parameters = {}
         parameters = load_optimization_parameters(self._parameter_yaml)
         for param in parameters:
-            trial_parameters[param.jaml_variable] = getattr(trial, param.optuna_method)(
-                **param.to_optuna_args()
-            )
+            trial_parameters[param.jaml_variable] = FlowOptimizer._suggest(param, trial)
 
         trial.workspace = self._workspace_base_dir + '/JINA_WORKSPACE_' + '_'.join([str(v) for v in trial_parameters.values()])
 
         return trial_parameters
 
+    @staticmethod
+    def _suggest(param, trial):
+
```

Comment:
what do you think about having something linke this in order to get rid of the redundant code?:
```
class_to_method = {
    IntegerParameter: trial.suggest_int,
    FloatParameter: trial.suggest_float,
    ...
}
return class_to_method[param.class](**param)
```

Reply:
This will be a bit more subtle. We have to also do the parameter mapping. and `**param` does not work, as long as the parameter is now dict. That would result in the same code as before with the `to_optuna_args` function, non?

Reply:
We could also use the `vars` build in like: `**vars(param)` 
As a down site other unwanted parameters are transferred as well and `name` is not defined. However, with some small refactoring on the parameters it might be possible.
```
vars(IntegerParameter(1,2,3,True)) == {
    'parameter_name': '', 
    'jaml_variable': 'JINA_NONE_', 
    'low': 1, 'high': 2, 'step_size': 3, 
    'log': True, 
    'optuna_method': 'suggest_int'
}
```

I just wanted to put this solution for discussion. There is no request for change from my site.

Reply:
Hmm I prefer stupid easily readable code above elegant non-readable code for such kind of parsing. There is no complexity in the `ifelse`, which would be circumvented when doing anything more elaborated.